test/alternator/test_streams.py

```diff
@@ -735,17 +781,95 @@ def do_test(test_table_ss_stream, dynamodbstreams, updatefunc, mode, p = random_
     pytest.fail('missing events in output: {}'.format(output))
 
 # Test a single PutItem of a new item. Should result in a single INSERT
-# event. Currently fails because in Alternator, PutItem - which generates a
-# tombstone to *replace* an item - generates REMOVE+MODIFY (issue #6930).
-@pytest.mark.xfail(reason="Currently fails - see issue #6930")
+# event. Reproduces #6930.
 def test_streams_putitem_keys_only(test_table_ss_keys_only, dynamodbstreams):
     def do_updates(table, p, c):
         events = []
         table.put_item(Item={'p': p, 'c': c, 'x': 2})
-        events.append(['INSERT', {'p': p, 'c': c}, None, {'p': p, 'c': c, 'x': 2}])
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'x': 2}])
         return events
     do_test(test_table_ss_keys_only, dynamodbstreams, do_updates, 'KEYS_ONLY')
 
+def test_streams_putitem_new_item_overrides_old_lsi(test_table_sss_new_and_old_images_lsi, dynamodbstreams):
+    def do_updates(table, p, c):
+        events = []
+        table.put_item(Item={'p': p, 'c': c, 'a': '1'})
+        # TODO: change to 'INSERT' when #6918 is fixed
+        events.append([['INSERT', 'MODIFY'], {'p': p, 'c': c}, None, {'p': p, 'c': c, 'a': '1'}])
+        table.put_item(Item={'p': p, 'c': c})
```

Comment:
Maybe you want to read the item at this point (from the base table, not CDC) to see the old LSI index ("a") was properly deleted - this was the bug you had before we moved the LSI key to :attrs.
Or, I remember now, you had a separate test for that in the previous patch? It probably belongs in this patch (but not in this test file)